### PR TITLE
Implements state_dict and load_state_dict functions for custom schedulers

### PR DESCRIPTION
**Pull request by nbrgr**
This commit adds the functions state_dict and load_state_dict
for the custom learning rate schedulers implemented in joeynmt.
Without these implemented, the learning rate scheduler restarts
when resuming from a checkpoint.

### CODE REVIEW

**Review by juliakreutzer — APPROVED** [review]
Cool! 💯
